internal/compiler/pathutils.rs

```diff
@@ -0,0 +1,528 @@
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+use std::path::{Path, PathBuf};
+
+// Reimplement some Path handling code: The one in `std` is not available
+// when running in WASM!
+//
+// This is not helped by us using URLs in place of paths *sometimes*.
+
+pub fn is_url(path: &Path) -> bool {
```

Comment:
This function and many other seems like internal details for clean_path.  It would be better not to make everything `pub` so it's easier to see what is "interface" and what isn't.
Ideally pub function also have documentation (and non-pub function as well)